sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/RetryUtil.java

```diff
@@ -51,51 +56,54 @@ public static AmqpRetryPolicy getRetryPolicy(AmqpRetryOptions options) {
      * Given a {@link Flux} will apply the retry policy to it when the operation times out.
      *
      * @param source The publisher to apply the retry policy to.
+     *
      * @return A publisher that returns the results of the {@link Flux} if any of the retry attempts are successful.
-     *         Otherwise, propagates a {@link TimeoutException}.
+     *     Otherwise, propagates a {@link TimeoutException}.
      */
-    public static <T> Flux<T> withRetry(Flux<T> source, Duration operationTimeout, AmqpRetryPolicy retryPolicy) {
-        return Flux.defer(() -> source.timeout(operationTimeout))
-            .retryWhen(Retry.withThrowable(errors -> retry(errors, retryPolicy)));
+    public static <T> Flux<T> withRetry(Flux<T> source, AmqpRetryOptions retryOptions, String timeoutMessage) {
+        return source.timeout(retryOptions.getTryTimeout())
```

Comment:
_This is for a larger discussion_

Should we guard timeouts on timeout periods that are zero or negative? Timeout schedules a task on the parallel scheduler by default to propagate an exception if it runs before an emission is made from the reactive stream. From this could an instantly scheduled exception operation result in side affect/unexpected behavior?

cc: @srnagar @JonathanGiles

Reply:
Timeout periods that are zero or negative are invalid inputs and should be validated when it's set. If the timeout is `null` then we should use a default (if applicable) or never timeout.

Reply:
In our existing logic for AmqpRetryOptions, if the duration is zero, then we don't retry at all. But negative is invalid.